src/query/ast/src/ast/expr.rs

```diff
@@ -318,6 +318,8 @@ pub enum BinaryOperator {
     BitwiseOr,
     BitwiseAnd,
     BitwiseXor,
+    // `=>` operator,
+    FatRArrow,
```

Comment:
It's better to add the named option syntax to `Expr::FunctionCall`'s `args` instead of `BinaryOperator`. The field `key` add to `Scalar::Constant` and the constraint of `left of => must be an identifier and the right must be literal` both reveal that the approach may  not be optimal.

Reply:
We just extract the name of the identifier and convert it into a constant string scalar.

Reply:
The problem is that the current ad-hoc design may add unnescessary complexity to the ast

Reply:
The optimal solution may be like:

```diff
  pub enum Expr<'a> {
      ....
      FunctionCall {
          span: &'a [Token<'a>],
          /// Set to true if the function is aggregate function with `DISTINCT`, like `COUNT(DISTINCT a)`
          distinct: bool,
          name: Identifier<'a>,
          args: Vec<Expr<'a>>,
+         named_args: HashMap<String, Expr<'a>>,
          params: Vec<Literal>,
      },
      ....
  } 
```